integration_tests/test_asset_recording.py

```diff
@@ -55,6 +56,35 @@ def test_prime_records_snapcraft_yaml_copy(self):
 
 class ManifestRecordingTestCase(AssetRecordingBaseTestCase):
 
+    def test_prime_records_uname(self):
+        self.run_snapcraft('prime', project_dir='basic')
+
+        recorded_yaml_path = os.path.join(
+            self.prime_dir, 'snap', 'manifest.yaml')
+        with open(recorded_yaml_path) as recorded_yaml_file:
+            recorded_yaml = yaml.load(recorded_yaml_file)
+
+        expected_uname = subprocess.check_output(
+            ['uname', '-a']).decode('utf-8').strip()
+        self.assertThat(
+            recorded_yaml['parts']['dummy-part']['uname'],
+            Equals(expected_uname))
+
+    def test_prime_records_installed_packages(self):
+        self.run_snapcraft('prime', project_dir='basic')
+
+        recorded_yaml_path = os.path.join(
+            self.prime_dir, 'snap', 'manifest.yaml')
+        with open(recorded_yaml_path) as recorded_yaml_file:
+            recorded_yaml = yaml.load(recorded_yaml_file)
+
+        with apt.Cache() as apt_cache:
+            expected_package = 'python3={}'.format(
+                apt_cache['python3'].installed.version)
```

Comment:
How about checking repo.get_installed_packages as well? My gut feeling is we should also explicitly test that it works correctly (even if the manifest uses it indirectly) - or if not here, in a unit test.

Reply:
I added a unit test in snapcraft/tests/repo/test_deb.py